### PR TITLE
Fix markdown tab render delay when toggling selected compared runs

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
JIRA: https://issues.redhat.com/browse/RHOAIENG-8360

## Description

This issue is because we have 2 additional S3 API calls for fetching the markdown content. In that way, we have a delay when selecting/unselecting the compared runs. The solution is to fetch all the markdown data on the first render and record it as a map. Then we only need to filter the map based on the run selection.

## How Has This Been Tested?

1. Turn S3 feature flag on
2. Create 2 runs with the pipeline provided in the ticket
3. Go to compare runs page
4. Select/Unselect the runs to see if the markdown tab is rendering seamlessly and correctly

## Test Impact

N/A

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by Gkrumbach07 — COMMENTED** [review]
code looks good, tested it live and works well

/lgtm

waiting on one more review

File: frontend/src/concepts/pipelines/content/compareRuns/metricsSection/markdown/useFetchMarkdownMaps.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+          );
+          const text = await fetchStorageObject(namespace, uriComponents.path).catch(() => null);
+
+          if (text === null) {
+            return null;
+          }
+
+          return { run, sizeBytes, text, path };
+        }),
+    [fullArtifactPaths, namespace],
```
Comment (christianvogt) [inline frontend/src/concepts/pipelines/content/compareRuns/metricsSection/markdown/useFetchMarkdownMaps.ts:57]: I see that this was also present in the old implementation but `fullArtifactPaths` is a very broad dependency that could trigger a lot of fetching when the fetching only needs to happen based on `path.linkedArtifact.artifact.getUri()`.

Comment (DaoDaoNoCode) [reply frontend/src/concepts/pipelines/content/compareRuns/metricsSection/markdown/useFetchMarkdownMaps.ts:57]: Created https://issues.redhat.com/browse/RHOAIENG-9453 to unblock this PR.

**Pull request merged**